### DISCUSSION TITLE
Update pimcore 5.7 to 5.8

### DISCUSSION THREAD

**Discussion by butlersupply** [pimcore/pimcore#12373] [Q&A / Support]
Using:

https://pimcore.com/docs/pimcore/5.8/Development_Documentation/Installation_and_Upgrade/Updating_Pimcore/index.html

I'm running COMPOSER_MEMORY_LIMIT=-1 composer update in the directory for my pimcore project.  But the update fails with errors (see below).  Any help would be greatly appreciated.  I'm trying to get to the latest version but this is the first step...

 [WARNING] Some commands could not be registered:

In DumpSitemapsCommand.php line 45:

  Type error: Too few arguments to function Presta\SitemapBundle\Command\Dump
  SitemapsCommand::__construct(), 2 passed in /var/www/html/mycompany/var/
  cache/prod/ContainerSe5h57z/getPrestaSitemap_DumpCommandService.php on line
   8 and exactly 3 expected

In MaintenanceCommand.php line 45:

  Type error: Too few arguments to function Pimcore\Bundle\CoreBundle\Command
  \MaintenanceCommand::__construct(), 0 passed in /var/www/html/mycompany/
  var/cache/prod/ContainerSe5h57z/getMaintenanceCommandService.php on line 8
  and exactly 2 expected

14:56:08 ERROR     [console] Error thrown while running command "--ansi cache:clear --no-warmup". Message: "Unable to write in the "/var/www/html/mycompany/var/cache/prod" directory." ["exception" => Symfony\Component\Console\Exception\RuntimeException { …},"command" => "--ansi cache:clear --no-warmup","message" => "Unable to write in the "/var/www/html/mycompany/var/cache/prod" directory."]

  Unable to write in the "/var/www/html/mycompany/var/cache/prod" director
  y.

cache:clear [--no-warmup] [--no-optional-warmers] [-h|--help] [-q|--quiet] [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interaction] [--ignore-maintenance-mode] [--maintenance-mode] [-e|--env ENV] [--no-debug] [--] <command>

Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the symfony-scripts event terminated with an exception

  [RuntimeException]
  An error occurred when executing the "'cache:clear --no-warmup'" command:

   [WARNING] Some commands could not be registered:

  In DumpSitemapsCommand.php line 45:

    Type error: Too few arguments to function Presta\SitemapBundle\Command\Du
  mp
    SitemapsCommand::__construct(), 2 passed in /var/www/html/mycompany/va
  r/
    cache/prod/ContainerSe5h57z/getPrestaSitemap_DumpCommandService.php on li
  ne
     8 and exactly 3 expected

  In MaintenanceCommand.php line 45:

    Type error: Too few arguments to function Pimcore\Bundle\CoreBundle\Comma
  nd
    \MaintenanceCommand::__construct(), 0 passed in /var/www/html/butlersuppl
  y/
    var/cache/prod/ContainerSe5h57z/getMaintenanceCommandService.php on line
  8
    and exactly 2 expected

  14:56:08 ERROR     [console] Error thrown while running command "--ansi cac
  he:clear --no-warmup". Message: "Unable to write in the "/var/www/html/butl
  ersupply/var/cache/prod" directory." ["exception" => Symfony\Component\Cons
  ole\Exception\RuntimeException { …},"command" => "--ansi cache:clear --no-w
  armup","message" => "Unable to write in the "/var/www/html/mycompany/var
  /cache/prod" directory."]

    Unable to write in the "/var/www/html/mycompany/var/cache/prod" direct
  or
    y.

  cache:clear [--no-warmup] [--no-optional-warmers] [-h|--help] [-q|--quiet]
  [-v|vv|vvv|--verbose] [-V|--version] [--ansi] [--no-ansi] [-n|--no-interact
  ion] [--ignore-maintenance-mode] [--maintenance-mode] [-e|--env ENV] [--no-
  debug] [--] <command>

update [--prefer-source] [--prefer-dist] [--dry-run] [--dev] [--no-dev] [--lock] [--no-custom-installers] [--no-autoloader] [--no-scripts] [--no-progress] [--no-suggest] [--with-dependencies] [--with-all-dependencies] [-v|vv|vvv|--verbose] [-o|--optimize-autoloader] [-a|--classmap-authoritative] [--apcu-autoloader] [--ignore-platform-reqs] [--prefer-stable] [--prefer-lowest] [-i|--interactive] [--root-reqs] [--] [<packages>]...

**Comment by jdreesen** [comment]
Which Composer version are you using? If it's v2, I think v2.2 (the LTS version) is the maximum that works, so if you're on the latest v2.3, try downgrading.

**Comment by butlersupply** [comment]
Thank you very much.  I'm still on v1 according to the one of the warnings.  Should I upgrade to 2.3 then?

**Loading composer repositories with package information
Warning from https://repo.packagist.org: Support for Composer 1 is deprecated and some packages will not be 
available. You should upgrade to Composer 2. See https://blog.packagist.com/deprecating-composer-1-support/**

**Reply by jdreesen** [reply]
> Should I upgrade to 2.3 then?

No, I'm pretty sure v2.3 won't work. Composer 1 is probably fine for now. Must be something else, then :/

**Reply by butlersupply** [reply]
Thank you for your help.

**Comment by aweichler** [comment]
It looks like there is a permission problem, when writing the cache. Do you have checked the permissions? But I think it is only one Problem.

**Reply by butlersupply** [reply]
Yes it looks like you are right about the permissions.  I tried:

	cd /var/www/html/mycompany/
	sudo chown -R www-data:www-data *
	sudo chmod -R 777 *
	COMPOSER_MEMORY_LIMIT=-1 composer update

And that failed with the message below.  Now running bin/console does nothing.  No help page or message of any kind.  

**> Pimcore\Composer::postUpdate
> Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache
Script Sensio\Bundle\DistributionBundle\Composer\ScriptHandler::clearCache handling the symfony-scripts event terminated with an exception

  [RuntimeException]
  An error occurred when executing the "'cache:clear --no-warmup'" command:**

This system is a copy of my DEV system.  I'm going to make a fresh copy and try again with

sudo -u www-data COMPOSER_MEMORY_LIMIT=-1 composer update
